osu.Game.Rulesets.Osu/Skinning/Legacy/LegacySpinner.cs

```diff
@@ -16,6 +16,9 @@ namespace osu.Game.Rulesets.Osu.Skinning.Legacy
 {
     public abstract class LegacySpinner : CompositeDrawable
     {
+        protected static readonly float SPINNER_TOP_OFFSET = MathF.Ceiling(45f * SPRITE_SCALE);
```

Comment:
The next piece of the puzzle that confuses me is this line. I don't really understand what the 0.625 scale is doing here, on a positional value, where we're already in a x480 coordinate space via `OsuPlayfieldAdjustmentContainer`.

It seems to me like this works out of pure luck, because `ceil(45 - 45 * 0.625) = 17`. however this is 1px off stable which subtracts 16 instead. Is there a reason why it was done this way?

@peppy Also pinging you because I've struggled with the 1.6/0.625 stuff over the years, so I might be missing something here.

Reply:
Without looking into it too deep, if it is aligning to sprites (ie. the 45 here is the pixels in the sprite?) then this will be converting to gamefield pixels. Sprites are always at a higher resolution.

It really depends on what 45 means in this context as to whether it's correct or not.

Reply:
All spinner sprites in stable are `Fields.TopLeft`, so the only conversion could be x480 <=> x768. However if it was a conversion from some sprite size in x768 to x480 for the purpose of sprite alignment, then I'd expect stable to be doing it as well, which it isn't. Is that a fair assertion?

Reply:
Put another way: if we're deviating from copying across the stable variables verbatim in any way, it should be explained inline in the code. So far we've been able to implement legacy items without doing this right?

I know that spinner may have had some weirdness with centering implementation on stable, but I'd have to go through this PR in detail to see if that's something we haven't already tackled in lazer-legacy implementation (I do plan on reviewing this one if it's not already merged... another day).

Reply:
That was quite a shocking coincidence (or pure luck maybe), there's indeed no sense of the spinner offset to be multiplied by `SPRITE_SCALE`.

Looking at the code snippet you posted previously [here](https://github.com/ppy/osu/pull/11975#discussion_r589150694):
```csharp
protected int spinnerTopOffset = 45;

if (GameBase.GameField.CorrectionOffsetActive)
    spinnerTopOffset -= 16;  // (this is the -8 in our code)
```

`SPINNER_TOP_OFFSET` must apply the `-16` offset always here in lazer, for [the same reason](https://github.com/ppy/osu/blob/05493958696d488c4a8b76582ee0f92d5e7cf75b/osu.Game.Rulesets.Osu/Skinning/Legacy/LegacySpinner.cs#L39-L41) why `LegacySpinner` applies a `-8` offset. (playfield shifting is handled at a higher level)

I will change the constant to do that with the proper reason why.

Apologies for pushing such change without any proper thinking of why it has to be done like that.